src/io/evented.cr

```diff
@@ -136,7 +136,7 @@ module IO::Evented
   end
 
   private def add_read_event(timeout = @read_timeout) : Nil
```

Comment:
These are used by among others the `evented_read` and `evented_write` methods defined here in `Evented`. However, those two look directly at `Errno.value ` which I believe isn't set by uring SQEs (it might be set by the uring syscall operations themselves, but that is not helpful, for obvious reasons). 

.. the things I discover the hard way by implementing the same thing, differently :D

Reply:
Syscalls never set errno, not even the uring operations. The errno global variable is a libc thing, and we must be sure not to use it.

A future PR implementing more SQEs would work as a fast path, entirely bypassing `evented_read`. So I don't think this will be an issue, even later. Something like this, but somehow refactored to be less repetitive:

```
module Crystal::System::FileDescriptor
  private def unbuffered_read(slice : Bytes)`
    if io_uring.available? && io_uring.read?
      ret = io_uring.read(fd, slice, slice.size, timeout: @read_timeout)

      if ret < 0
        err = Errno.new(-ret)
        if err == Errno::EBADF
          raise IO::Error.new "File not open for reading"
        elsif err == Errno::ECANCELED
          raise IO::TimeoutError.new("Read timed out")
        else
          raise IO::Error.from_os_error("Error reading file", err)
        end
      end

      ret
    else
      evented_read(slice, "Error reading file") do
        LibC.read(fd, slice, slice.size).tap do |return_code|
          if return_code == -1 && Errno.value == Errno::EBADF
            raise IO::Error.new "File not open for reading"
          end
        end
      end
	end
  end
end
```